Doc/extending/extending.rst

```diff
@@ -222,7 +222,12 @@ with an exception object (leaving out the error checking for now)::
 
        SpamError = PyErr_NewException("spam.error", NULL, NULL);
        Py_INCREF(SpamError);
-       PyModule_AddObject(m, "error", SpamError);
+       if (PyModule_AddObject(m, "error", SpamError)) {
+           Py_DECREF(SpamError);
```

Comment:
```suggestion
           Py_XDECREF(SpamError);
```
Actually it may need
```c
           Py_XDECREF(SpamError);
           Py_CLEAR(SpamError);
```
but this is a different issue.